pkg/manifest/yaml/resources.go

```diff
@@ -0,0 +1,59 @@
+package yaml
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+)
+
+// WriteResources takes a prefix path, and a map of paths to values, and will
+// marshal the values to the filenames as YAML resources, joining the prefix to
+// the filenames before writing.
+//
+// It returns the list of filenames written out.
+func WriteResources(path string, files map[string]interface{}) ([]string, error) {
+	filenames := make([]string, 0)
+	for filename, item := range files {
+		err := marshalItemsToFile(filepath.Join(path, filename), list(item))
+		if err != nil {
+			return nil, err
+		}
+		filenames = append(filenames, filename)
+	}
+	return filenames, nil
+}
+
+func marshalItemsToFile(filename string, items []interface{}) error {
+	err := os.MkdirAll(filepath.Dir(filename), 0755)
+	if err != nil {
+		return fmt.Errorf("failed to MkDirAll for %s: %v", filename, err)
+	}
+	f, err := os.Create(filename)
+	if err != nil {
+		return fmt.Errorf("failed to Create file %s: %v", filename, err)
+	}
+	defer f.Close()
+	return marshalOutputs(f, items)
```

Comment:
`return marshalOutput(f, item)`

Reply:
marshalOutputs allows for writing more than one resource to a single file.

This can be useful (for things like role-bindings, where you likely don't want to have a file _per_ rolebinding).